src/topology/metric_space/basic.lean

```diff
@@ -863,6 +863,13 @@ lemma closed_ball_Icc {x r : ℝ} : closed_ball x r = Icc (x-r) (x+r) :=
 by ext y; rw [mem_closed_ball, dist_comm, real.dist_eq,
   abs_sub_le_iff, mem_Icc, ← sub_le_iff_le_add', sub_le]
 
+-- This lemma could go somewhere else, but I did not find a better fit for it.  Feel
+-- free to suggest where it may be added.  Here, all the needed imports are available!
+lemma mem_Icc_iff_abs_le {R : Type*} [linear_ordered_add_comm_group R] {x y z : R} :
+  abs (x - y) ≤ z ↔ y ∈ Icc (x - z) (x + z) :=
+⟨λ h, ⟨sub_le.mp (abs_le.mp h).2, neg_le_sub_iff_le_add.mp (abs_le.mp h).1⟩,
+ λ hy, abs_le.mpr ⟨neg_le_sub_iff_le_add.mpr hy.2, sub_le.mp hy.1⟩⟩
```

Comment:
A shorter proof is
```suggestion
abs_le.trans ⟨
  λ h, ⟨sub_le.mp h.2, neg_le_sub_iff_le_add.mp h.1⟩,
  λ hy, ⟨neg_le_sub_iff_le_add.mpr hy.2, sub_le.mp hy.1⟩⟩
```

Reply:
Thanks for the proof!  Committed!

Reply:
Note the intermediate iff, `x - y ∈ set.Icc (-z) z ↔ y ∈ set.Icc (x - z) (x + z)`, is probably worth its own lemma somewhere too, along with the `+` form.

Reply:
That is a good suggestion, but I would not clutter this PR further: I am having a hard time getting the Liouville results getting merged as is, that I prefer to maintain the PRs as direct as possible.  If you want, feel free to PR this yourself.  I had also imagined that various Ico, Ioc, Ioo,... versions would be in the same spirit, but I did not need them for my application.

Reply:
Done in #6202